tensorflow-op-codegen/src/parser.rs

```diff
@@ -155,6 +157,40 @@ fn data_type<'a>(input: &'a [u8]) -> ParseResult<'a, DataType> {
     })(input)
 }
 
+fn type_id<'a>(input: &'a [u8]) -> ParseResult<'a, FullTypeId> {
+    map_res(identifier, |s| match &s as &str {
```

Comment:
I know this is how we do it above, but this would be more robust if parsed with something like:

```rust
map_res(identifier, |s| {
    match FullTypeId::values()
        .iter()
        .filter(|d| d.descriptor().name() == s)
        .next()
    {
        Some(d) => Ok(FullTypeId::from_i32(d.value()).unwrap()),
        None => Err(nom::Err::<VerboseError<&'a [u8]>>::Error(make_error(
            input,
            ErrorKind::Alt,
        ))),
    }
})(input)
```

That way, if new enum values are added in the future, this code doesn't need to be updated.

(Ideally this would use `EnumDescriptor::value_by_name`, but that panics on error rather than returning a `Result` or `Option` for some reason.)

Reply:
Thanks for the review. I have changed the code to the one you suggested.

Shall I change `fn data_type` as well?